### PR TITLE
Add crud features to registration cli

### LINKED ISSUE #234: Registration CLI should provide basic CRUD operation
Issue #217 deals with making registration entries easier to create using the CLI, however we now need the same work done for the other operations. At the very least, one should be able to Create, Read/List, and Delete registration entries using the CLI tool.

The current Register subcommand is not positioned well for this, so perhaps that means the introduction of a new set of subcommands like `entry create` `entry show` etc.

### PR DESCRIPTION
**Pull request by evan2645**
Sorry for sitting on this for so long... the scope creeped like crazy.

This PR:
* Adds Create/Show/Delete functionality to registration CLI tool
* Renames/restructures registration CLI tool to make sense w/ CRUD features
* Updates Registration API to fill in logic for select by
* Refactors/fixes bug in datastore logic for list by selectors
* Extends datastore and registration api to allow for fetching of all entries

Fixes #234

### CODE REVIEW

File: cmd/spire-server/cli/entry/show.go [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+	s.filterEntries()
+	s.printEntries()
+	return 0
+}
+
+func (s *ShowCLI) fetchAllEntries() error {
+	var err error
+	entries, err := s.Client.FetchEntries(context.TODO(), &common.Empty{})
+	if err != nil {
+		return nil
```
Comment (amartinezfayo) [inline cmd/spire-server/cli/entry/show.go]: Should be `return err`

**Comment by evan2645** [comment]
oops, good catch @amartinezfayo! fixed now

File: cmd/spire-server/cli/entry/delete.go [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+
+func (d DeleteCLI) Help() string {
+	_, err := d.newConfig([]string{"-h"})
+	return err.Error()
+}
+
+func (d DeleteCLI) Run(args []string) int {
+	config, err := d.newConfig(args)
+	if err != nil {
+		fmt.Println(err.Error())
```
Comment (amartinezfayo) [inline cmd/spire-server/cli/entry/delete.go]: Looking at how printing the error and returning 1 is repeated multiple times, I was thinking that maybe we can have this in a separate function? Maybe that could help to make consistent changes in that logic if needed in the future.

**Comment by evan2645** [comment]
@amartinezfayo fixed! thanks for the fast CR

**Pull request merged**